src/toc.tsx

```diff
@@ -1,47 +1,82 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
 import { ActivityMonitor, PathExt } from '@jupyterlab/coreutils';
-
 import { IDocumentManager } from '@jupyterlab/docmanager';
-
 import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
-
 import { Message } from '@phosphor/messaging';
-
 import { Widget } from '@phosphor/widgets';
+import { IHeading } from './utils/headings';
+import { Registry } from './registry';
+import { TOCTree } from './toc_tree';
 
-import { TableOfContentsRegistry } from './registry';
+/**
+ * Interface describing table of contents widget options.
+ *
+ * @private
+ */
+interface IOptions {
+  /**
+   * Application document manager.
+   */
+  docmanager: IDocumentManager;
 
-import * as React from 'react';
-import * as ReactDOM from 'react-dom';
+  /**
+   * Application rendered MIME type.
+   */
+  rendermime: IRenderMimeRegistry;
+}
 
 /**
- * Timeout for throttling TOC rendering.
+ * Interface describing the current widget.
+ *
+ * @private
```

Comment:
Why have you marked this as private? It is also part of the public interface. I think it would be best to have an `ITableOfContents` namespace for the this and the above `IOptions`. This is mostly equivalent to static class attributes, but more in keeping with previous code style,

Reply:
Removed. Re: namespace. Not of the opinion that this is necessary, as, e.g., in an editor like VSCode, the main intent is to describe the "shape" of data, rather than defining a type to which one casts. Removing the `@private` annotations here makes sense from the perspective of using `typedoc` and allowing readers to inspect input and output data shapes. So went with just removing the `@private` annotations.

Reply:
I agree that it is not necessary, it is more of an issue of following a consistent style and set of design patterns which is used in JupyterLab. See, for instance, the same pattern [here](https://github.com/jupyterlab/jupyterlab/blob/f2e0cde0e7c960dc82fd9b010fcd3dbd9e9b43d0/packages/docmanager/src/manager.ts#L593-L639).

Reply:
I'm actually a little surprised this compiles: these interfaces should be public and usable by other extensions, but are not exported by this module.

Reply:
There may be some confusion here as to what [`@private`](https://jsdoc.app/tags-private.html) does. This annotation originates from JSDoc and does not affect "usability" in any sense, which is in contrast to TypeScript's notion of [`private`](https://www.typescriptlang.org/docs/handbook/classes.html#understanding-private) class members. The code compiled because it is not necessary to export them or have them be annotated as public, as they simply describe the shape of data and are compiled away. The original reason I added the `@private` annotation was in (mistakingly) thinking that they would not be needed displayed in generated TypeDoc documentation. Removing the annotations allows them to be.

Regardless, if the desired convention is to export them as part of a namespace, the above is a moot point.